### PR TITLE
[@types/websocket] update definitions for 1.0.30 release

### LINKED ISSUE #30534: Wrong type for headers in w3cwebsocket constructor
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9acca8ab3e8b198aa3332f3ff6f03c0a6e3ce1e9/types/websocket/index.d.ts#L697

headers?: any[] should be headers?: any

headers passed as dictionary and not as list to the constructor

### PR DESCRIPTION
**Pull request by MasterOdin**
The previous type definition was out of date and lacking a fair number of methods, parameters, etc. This updates the bindings to match the associated features in the current release, what events get emitted, etc.

Additionally, this updates the tslint rules, removing all customizations and making it match the expected standard style.

Closes #30534

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/theturtle32/WebSocket-Node
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/websocket/index.d.ts [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
 
-
 import events = require('events');
 import http = require('http');
 import https = require('https');
 import net = require('net');
 import url = require('url');
 
-export interface IStringified {
+export interface Stringified {
```
Comment (armanio123) [inline types/websocket/index.d.ts]: I suggest not renaming the interfaces as they introduce a breaking change for people importing them.

**Review by armanio123 — COMMENTED** [review]
I ping the owners for review as this is a fairly big change and I'm not familiar with the library.

File: types/websocket/index.d.ts (hunk 1 above)
Comment (MasterOdin) [reply types/websocket/index.d.ts]: Makes sense, reverted all interfaces to have the leading `I` again and ignored the appropriate tslint rule.

**Review by andrewbranch — APPROVED** [review]
Awesome, thanks!

**Pull request merged**